step.go

```diff
@@ -261,56 +257,42 @@ func (s BuildForSimulatorStep) Run(cfg RunOpts) (ExportOptions, error) {
 		return ExportOptions{}, fmt.Errorf("failed to get absolute project path: %s", err)
 	}
 
-	//
-	// Read the Scheme
-	var scheme *xcscheme.Scheme
-	var schemeContainerDir string
-	var conf string
-	{
-		scheme, schemeContainerDir, err = readScheme(absProjectPath, cfg.Scheme)
-
-		if err != nil {
-			return ExportOptions{}, fmt.Errorf("failed to read schema: %s", err)
-		}
-
-		if cfg.Configuration != "" {
-			conf = cfg.Configuration
-		} else {
-			conf = scheme.ArchiveAction.BuildConfiguration
-		}
+	tmpDir, err := pathutil.NormalizedOSTempDirPath("xcodeArchive")
+	if err != nil {
+		return ExportOptions{}, fmt.Errorf("failed to create temp dir, error: %s", err)
 	}
-
-	//
-	// Create the app with Xcode Command Line tools
+	archivePth := filepath.Join(tmpDir, cfg.Scheme+"-simulator.xcarchive")
 	{
 		fmt.Println()
 		log.Infof("Running build")
 
-		actions := []string{"build"}
+		actions := []string{"archive"}
 		if cfg.PerformCleanAction {
 			actions = append(actions, "clean")
 		}
 
-		// Build for simulator command
-		xcodeBuildCmd := xcodebuild.NewCommandBuilder(absProjectPath, actions...)
-		xcodeBuildCmd.SetScheme(cfg.Scheme)
-		xcodeBuildCmd.SetConfiguration(conf)
-		xcodeBuildCmd.SetDestination(cfg.Destination)
-		xcodeBuildCmd.SetCustomOptions(cfg.XcodebuildAdditionalOptions)
+		archiveCmd := xcodebuild.NewCommandBuilder(absProjectPath, actions...)
+		archiveCmd.SetArchivePath(archivePth)
+		archiveCmd.SetScheme(cfg.Scheme)
+		if cfg.Configuration != "" {
+			archiveCmd.SetConfiguration(cfg.Configuration)
+		}
+		archiveCmd.SetDestination(cfg.Destination)
+		archiveCmd.SetCustomOptions(cfg.XcodebuildAdditionalOptions)
 		if cfg.XCConfigContent != "" {
 			xcconfigPath, err := s.XCConfigWriter.Write(cfg.XCConfigContent)
 			if err != nil {
 				return ExportOptions{}, fmt.Errorf("failed to write xcconfig file contents: %w", err)
 			}
-			xcodeBuildCmd.SetXCConfigPath(xcconfigPath)
+			archiveCmd.SetXCConfigPath(xcconfigPath)
 		}
 
 		swiftPackagesPath, err := cache.SwiftPackagesPath(absProjectPath)
 		if err != nil {
 			return ExportOptions{}, fmt.Errorf("failed to get Swift Packages path: %s", err)
 		}
 
-		rawXcodeBuildOut, err := runCommandWithRetry(xcodeBuildCmd, cfg.LogFormatter == "xcpretty", swiftPackagesPath)
+		rawXcodeBuildOut, err := runCommandWithRetry(archiveCmd, cfg.LogFormatter == "xcpretty", swiftPackagesPath)
```

Comment:
Not strictly PR scope, but could support xcbeautify here (as in the xcode-test step)